### PR TITLE
MM-40013 : Mock useIntl in all test cases

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-40013

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note

```

### CODE REVIEW

File: components/post_view/post_aria_label_div.test.jsx [hunk 1]
```diff
@@ -56,16 +47,12 @@ describe('PostAriaLabelDiv', () => {
...
-        useIntl.mockImplementation(() => createIntl({locale: 'en', messages: enMessages, defaultLocale: 'en'}));
-
         let wrapper = mount(<PostAriaLabelDiv {...baseProps}/>, mountOptions);
         let div = wrapper.childAt(0);
 
         expect(div.prop('aria-label')).toContain(author.username);
         expect(div.prop('aria-label')).toContain('January');
 
-        useIntl.mockImplementation(() => createIntl({locale: 'es', messages: esMessages, defaultLocale: 'es'}));
-
```
Comment (M-ZubairAhmed) [inline components/post_view/post_aria_label_div.test.jsx:68]: Hey @hmhealey i am not able to restore mock to `en` here, lets discuss

**Comment by M-ZubairAhmed** [comment]
Point to note here is however `useIntl` is mocked with default english translations, if we require other language then in those file we need to override the react-intl with the new language.

**Review by koox00 — APPROVED** [review]

**Review by michelengelen — APPROVED** [review]
LGTM ... nice catch and simplification! ✨

File: tests/react-intl_mock.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+jest.mock('react-intl', function() {
+    const reactIntl = jest.requireActual('react-intl');
+    const enMessages = require('i18n/es.json');
```
Comment (hmhealey) [inline tests/react-intl_mock.js:6]: Looks mostly good, but you're loading the Spanish translation strings by mistake
```suggestion
    const enMessages = require('i18n/en.json');
```

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: tests/react-intl_mock.js (hunk 2 above)
Comment (M-ZubairAhmed) [reply tests/react-intl_mock.js]: Ops done

**Review by hmhealey — APPROVED** [review]

**Pull request merged**